tests/unittests/GraphOptzTest.cpp

```diff
@@ -372,6 +372,11 @@ TEST_F(GraphOptz, optimizeBatchNormAfterConvWithPred) {
   ASSERT_EQ(newCV->getNumUsers(), 1);
   Node *save = newCV->getUsers().begin()->getUser();
   EXPECT_TRUE(llvm::isa<SaveNode>(save));
+
+  optimizedF_ = optimizeFunction(F_);
```

Comment:
The numerical equivalence check feeds the same input to both `F_` and `optimizedF_`. `F_` at this point is already optimized due to line 277, so the numerical equivalence check is not useful. You should not optimize `F_` - this is where the `optimizeFunction` helper comes into play. It clones the function first, and then optimizes the clone.

Reply:
Thank you for your clarification! I tried to change my code according to your advice, however, I still encounter one failure in line 374: ASSERT_TRUE(newCV->hasPredicate());
At your convenience, would you mind giving me some hint? The attached image is the modified version of my code.
<img width="715" alt="Screen Shot 2019-10-22 at 9 57 51 PM" src="https://user-images.githubusercontent.com/56737462/67359033-106ad300-f517-11e9-970c-07faf0b15bcc.png">

Reply:
That's weird. I recommend printing out the graph using `Function::dumpDAG` and looking at it with `graphviz` to see what the cloned graph looks like.